src/test/java/fi/aalto/cs/apluscourses/presentation/base/SelectableNodeViewModelTest.java

```diff
@@ -101,25 +99,30 @@ public void testApplyFilterWhenChildReturnsTrue() {
     when(filter.apply(child1)).thenReturn(Optional.of(true));
     when(filter.apply(child2)).thenReturn(Optional.empty());
 
-    assertTrue(node.applyFilter(filter));
+
+    Optional<Boolean> result = node.applyFilter(filter);
+    assertTrue(result.isPresent());
+    assertTrue(result.get());
```

Comment:
Perhaps add the no-sonar-comments to these since sonar is too stupid to recognize the previous call to `isPresent`